### PR TITLE
(SERVER-2063) Collect the simulation.log.gz from runs

### PR DESCRIPTION
**Pull request by haus**
This commit begins archiving the simulation.log.gz from each run. In
order to do this effectively we also clean up the simulation logs from
the workspace after collection.

### CODE REVIEW

**Comment by haus** [comment]
I have a test run of this going, I'll report back tomorrow.

File: jenkins-integration/jenkins-jobs/common/scripts/jenkins/pipeline.groovy [hunk 1]
```diff
@@ -367,6 +367,12 @@ def step900_collect_driver_artifacts() {
     // NOTE: this DSL step requires the puppet-gatling-jenkins plugin.  It also
     // depends on some data that gets created via 025_collect_facter_data.sh
     puppetGatlingArchive()
+
+    // Always archive the simulation.log.gz from the run.
+    archive "simulation-runner/results/**/*.log.gz"
+    dir "simulation-runner/results" {
+        deleteDir
```
Comment (lindboe) [inline jenkins-integration/jenkins-jobs/common/scripts/jenkins/pipeline.groovy]: Is this really necessary if the box will just be razored later?

Comment (haus) [reply jenkins-integration/jenkins-jobs/common/scripts/jenkins/pipeline.groovy]: this is on the jenkins machine.

Comment (lindboe) [reply jenkins-integration/jenkins-jobs/common/scripts/jenkins/pipeline.groovy]: oh, then I have no idea what archive does.

Does this still allow the gatling reports to be displayed in jenkins?

Comment (haus) [reply jenkins-integration/jenkins-jobs/common/scripts/jenkins/pipeline.groovy]: i stuck it in `step900_collect_driver_artifacts` as that seemed like the right place for this.

Comment (lindboe) [reply jenkins-integration/jenkins-jobs/common/scripts/jenkins/pipeline.groovy]: also related question: if puppetGatlingArchive() doesn't do this, what does it do

Comment (haus) [reply jenkins-integration/jenkins-jobs/common/scripts/jenkins/pipeline.groovy]: I believe the reports are served from the `jobs/$job/builds/$build_number` directory, and this operates in the `jobs/$job/workspace` area. I'll know for sure after the current run completes.

Comment (haus) [reply jenkins-integration/jenkins-jobs/common/scripts/jenkins/pipeline.groovy]: puppetGatlingArchive seems to be the magic to inline the html report. the important bits seem to be here: https://github.com/puppetlabs/puppet-gatling-jenkins-plugin/blob/be8328bc9d39c638cf0a123beb8a3ec5bfe98c2e/src/main/java/com/puppetlabs/jenkins/plugins/puppetgatling/PuppetGatlingPublisher.java

Comment (lindboe) [reply jenkins-integration/jenkins-jobs/common/scripts/jenkins/pipeline.groovy]: Ah, okay. If you don't mind I'd like to wait for that to be sure, don't want to delete this and have it break for a first time run of anything

Comment (haus) [reply jenkins-integration/jenkins-jobs/common/scripts/jenkins/pipeline.groovy]: Waiting is fine. I'm pretty confident based on poking around in the jenkins FS.

**Comment by haus** [comment]
You can see this in action here: http://sclb4agkxecbfvc.delivery.puppetlabs.net:8080/job/latest-test-changes/

File: jenkins-integration/jenkins-jobs/common/scripts/jenkins/pipeline.groovy [hunk 2]
```diff
@@ -367,6 +367,12 @@ def step900_collect_driver_artifacts() {
     // NOTE: this DSL step requires the puppet-gatling-jenkins plugin.  It also
     // depends on some data that gets created via 025_collect_facter_data.sh
     puppetGatlingArchive()
+
+    // Always archive the simulation.log.gz from the run.
+    archive "simulation-runner/results/**/*.log.gz"
+    dir('simulation-runner/results') {
```
Comment (lindboe) [inline jenkins-integration/jenkins-jobs/common/scripts/jenkins/pipeline.groovy]: SINGLE QUOTES :D

Comment (haus) [reply jenkins-integration/jenkins-jobs/common/scripts/jenkins/pipeline.groovy]: NOTHING MAKES ANY SENSE

**Pull request merged**